File: Week 3/03. Nodes/ros2_ws/src/ros2_cpp/src/oop_node.cpp
#include "rclcpp/rclcpp.hpp"

using namespace std::chrono_literals;

class MyNode : public rclcpp::Node
{
public:
    MyNode() : Node("cpp_node"), counter_ {0}
    {
        RCLCPP_INFO(this->get_logger(), "Hello from C++ OOP node");
        timer_ = this->create_wall_timer(1s, std::bind(&MyNode::timer_callback, this));
    }

private:
    void timer_callback()
    {
        counter_ ++;
        RCLCPP_INFO(this->get_logger(), "Hello %d",counter_);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: Week 3/03. Nodes/ros2_ws/install/ros2_py_cpp/lib/ros2_py_cpp/oop_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__("my_node")
        self.counter_ = 0
        self.get_logger().info("Hello from ROS OOP node...")
        self.create_timer(0.55, self.timer_callback)

    def timer_callback(self):
        self.counter_ += 1
        self.get_logger().info(f"Hello {self.counter_}")

def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()